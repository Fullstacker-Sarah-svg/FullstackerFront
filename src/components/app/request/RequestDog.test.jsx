import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RequestDog from './RequestDogs';

describe('Renders the request dog page', () => {
  afterEach(() => cleanup());
    

  it('diplays the page for dogs', () => {
    const { asFragment } = render(<RequestDog
      name="daisy" 
      type="bulldog" 
      characteristic="brintle"
      onChange={jest.fn()}
      onSubmit={jest.fn()}  
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
