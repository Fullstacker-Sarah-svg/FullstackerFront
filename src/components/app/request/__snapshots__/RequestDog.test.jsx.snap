// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the request dog page diplays the page for dogs 1`] = `
<DocumentFragment>
  <form>
    <title
      date-testid="Dog"
    >
      Dog
    </title>
    <h3>
      Please select one
    </h3>
    <div>
      <input
        id="post"
        name="method"
        placeholder="Enter post"
        type="radio"
        value="post"
      />
      <label
        for="post"
      >
        Add an Dog
      </label>
      <input
        name="name"
        placeholder="Enter name"
        type="text"
        value="daisy"
      />
      <input
        name="type"
        placeholder="Enter type"
        type="text"
        value="bulldog"
      />
      <input
        name="characteristic"
        placeholder="Enter characteristic"
        type="text"
        value="brintle"
      />
    </div>
    <div>
      <input
        id="get"
        name="method"
        type="radio"
        value="get"
      />
      <label
        for="get"
      >
        Get all Dogs
      </label>
    </div>
    <div>
      <input
        id="find"
        name="method"
        type="radio"
        value="find"
      />
      <label
        for="find"
      >
        Find Dog by Id
      </label>
      <input
        name="id"
        placeholder="Enter ID of Dog"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        id="put"
        name="method"
        type="radio"
        value="put"
      />
      <label
        for="put"
      >
        Update dog by Id
      </label>
      <input
        name="id"
        placeholder="Enter ID of Dog"
        type="text"
        value=""
      />
      <br />
      <input
        name="name"
        placeholder="Update name"
        type="text"
        value="daisy"
      />
      <input
        name="type"
        placeholder="Update type"
        type="text"
        value="bulldog"
      />
      <input
        name="characteristic"
        placeholder="Update characteristic"
        type="text"
        value="brintle"
      />
    </div>
    <div>
      <input
        id="delete"
        name="method"
        type="radio"
        value="delete"
      />
      <label
        for="delete"
      >
        Delete Dog by Id
      </label>
      <input
        name="id"
        placeholder="Enter ID of Dog"
        type="text"
        value=""
      />
    </div>
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
